python/pyspark/sql/dataframe.py

```diff
@@ -347,13 +347,30 @@ def show(self, n=20, truncate=True, vertical=False):
          name | Bob
         """
         if isinstance(truncate, bool) and truncate:
-            print(self._jdf.showString(n, 20, vertical))
+            print(self._jdf.showString(n, 20, vertical, False))
         else:
-            print(self._jdf.showString(n, int(truncate), vertical))
+            print(self._jdf.showString(n, int(truncate), vertical, False))
 
     def __repr__(self):
         return "DataFrame[%s]" % (", ".join("%s: %s" % c for c in self.dtypes))
```

Comment:
I agree that it would be better to respect `spark.sql.repr.eagerEval.enabled` here as well.

Reply:
Thanks for your reply, this implement in 94f3414.